### PR TITLE
Change the package.Spec.Config field to a string

### PR DESCRIPTION
**Pull request by lewisdiamond**
The string will be parsed as YAML and used as values for packages.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: api/testdata/test.yaml [hunk 1]
```diff
@@ -5,7 +5,11 @@ metadata:
...
-    secret.bar: eolus
-    pvc.storage: 1Mi
+  config: |
+    config:
+      - foo: windom
+      - baz: baz
+    secret:
+      bar: eolus
+    pvc:
+      storage: 1Mi
```
Comment (TerryHowe) [inline api/testdata/test.yaml:15]: I haven't looked at the code in general, but I like the general idea of this although I think we should attempt to keep the yaml flat. The helm best practices  of keeping values.yaml flat as possible I think also would help our situation

Comment (lewisdiamond) [reply api/testdata/test.yaml:15]: We'd have to try to enforce that on each chart we produce. In general most charts keep it only 1-3 levels deep.

**Review by TerryHowe — APPROVED** [review]

File: api/v1alpha1/package.go [hunk 2]
```diff
@@ -19,32 +17,6 @@ func (config *Package) ExpectedKind() string {
...
-				}
-			}
-			newMap := make(map[string]interface{})
-			subMap[name] = newMap
-			subMap = newMap
-		}
-		subMap[key] = v
-	}
-	return mapInterfaces, nil
+	err = yaml.Unmarshal([]byte(config.Spec.Config), &mapInterfaces)
```
Comment (ewollesen) [inline api/v1alpha1/package.go:20]: We should probably wrap this error to add a little context.

**Pull request merged**